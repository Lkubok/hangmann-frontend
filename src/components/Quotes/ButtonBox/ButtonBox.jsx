import React, { PureComponent } from "react";
import { connect } from "react-redux";
import { deleteQuote } from "../../../actions/quoteActions";
import "./ButtonBox.scss";
import { Link } from "react-router-dom";
const { REACT_APP_API_HOST } = process.env;

export class ButtonBox extends PureComponent {
  handleDelete = id => () => {
    this.props.deleteQuote(REACT_APP_API_HOST, id, this.props.token);
  };
  render() {
    return (
      <div className="btn-box">
        <Link to={`/quotes/${this.props.id}/edit`} className="btn btn-edit">
          Edit
        </Link>
        {this.props.canDelete ? (
          <button
            className="btn btn-danger"
            onClick={this.handleDelete(this.props.id)}
          >
            Delete
          </button>
        ) : (
          <button className="btn btn-disabled" disabled={true}>
            Delete
          </button>
        )}
      </div>
    );
  }
}

const mapStateToProps = (state, ownProps) => ({
  id: ownProps.id,
  token: state.appParamsReducer.jwt
});

const mapDispatchToProps = {
  deleteQuote
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ButtonBox);
