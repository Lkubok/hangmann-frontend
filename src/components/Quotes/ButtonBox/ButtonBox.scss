@import "../../../scss/mixins.scss";
@import "../../../scss/variables.scss";

.btn-box {
  flex-basis: 100%;
  @media all and (max-width: 1920px) {
    @include flex-cc-row;
  }
  @media all and (max-width: 768px) {
    @include flex-cc-row;
    // align-items: stretch;
    width: 100%;
  }
}

.btn {
  @include flex-item-btn;
  flex-basis: 100%;
  // width: 100%;
  border: 0;
  border-radius: 5px;
  transition: all 0.4s ease-in-out;
  color: $white-color;
  font-size: 12px;
  &-edit {
    border: 1px solid $primary-color;
    &:hover {
      cursor: pointer;
      background-color: $primary-color-hover;
    }
  }
  &-danger {
    border: 1px solid $danger-color-light;
    &:hover {
      cursor: pointer;
      background-color: red;
    }
  }
  &-disabled{
    background-color: $dark-color-light;
    color: $dark-color-light;
    opacity: 0;
    pointer-events: none;
    // display: none;
    // &:hover {
    //   pointer-events: none;
    //   // cursor: initial;
    //   background-color: $dark-color;
    // }
  }
}
a{
  text-decoration: none;
}
