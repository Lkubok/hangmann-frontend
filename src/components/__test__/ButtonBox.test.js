import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { ButtonBox } from "../Quotes/ButtonBox/ButtonBox";
Enzyme.configure({ adapter: new Adapter() });

describe("ButtonBox component", () => {
  let props;
  const spyFunction = jest.fn();
  beforeEach(() => {
    props = {
      canDelete: true,
      id: "123",
      token: "123token",
      deleteQuote: spyFunction
    };
  });
  it("Should match snapshot", () => {
    const wrapper = shallow(<ButtonBox {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it("Should launch handleDelte when clicking Delete Button", () => {
    const wrapper = shallow(<ButtonBox {...props} />);
    wrapper.find(".btn-danger").simulate("click");
    expect(spyFunction).toHaveBeenCalled();
  });
  xit("Delete function should be called with 2 args", () => {
    const wrapper = shallow(<ButtonBox {...props} />);
    wrapper.find(".btn-danger").simulate("click");
    expect(spyFunction).toHaveBeenCalledWith(
      "http://localhost:3001/api",
      "123",
      "123token"
    );
  });
  it("Shouldn't allow to launch handleDelete when have no auth", () => {
    const spyFunction = jest.fn();
    let propsNonDelete = {
      canDelete: false,
      id: "123",
      token: "123token",
      deleteQuote: spyFunction
    };
    const wrapper = shallow(<ButtonBox {...propsNonDelete} />);
    wrapper.find(".btn-disabled").simulate("click");
    expect(spyFunction).not.toHaveBeenCalled();
  });
});
