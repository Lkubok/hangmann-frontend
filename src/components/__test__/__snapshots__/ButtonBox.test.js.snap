// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBox component Should match snapshot 1`] = `
<div
  className="btn-box"
>
  <Link
    className="btn btn-edit"
    to="/quotes/123/edit"
  >
    Edit
  </Link>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
